Formula/gajim.rb

```diff
@@ -0,0 +1,657 @@
+    sha256 "45daf020b370bda13a1429c859fcdff0b766c0576844211446f9266cae97fb0e"
+  end
+
+  resource "qrcode" do
+    url "https://files.pythonhosted.org/packages/19/d5/6c7d4e103d94364d067636417a77a6024219c58cd6e9f428ece9b5061ef9/qrcode-6.1.tar.gz"
+    sha256 "505253854f607f2abf4d16092c61d4e9d511a3b4392e60bff957a68592b04369"
+  end
+
+  resource "six" do
+    url "https://files.pythonhosted.org/packages/dd/bf/4138e7bfb757de47d1f4b6994648ec67a51efe58fa907c1e11e350cddfca/six-1.12.0.tar.gz"
+    sha256 "d16a0141ec1a18405cd4ce8b4613101da75da0e9a7aec5bdd4fa804d0e0eba73"
+  end
+
+  resource "soupsieve" do
+    url "https://files.pythonhosted.org/packages/fb/9e/2e236603b058daa6820193d4d95f4dcfbbbd0d3c709bec8c6ef1b1902501/soupsieve-1.9.1.tar.gz"
+    sha256 "b20eff5e564529711544066d7dc0f7661df41232ae263619dede5059799cdfca"
+  end
+
+  # some popular plugins
+
+  resource "pybonjour" do
+    url "https://dev.gajim.org/lovetox/pybonjour-python3/-/archive/e7c010db6372559b9ee9cb64dde75206db684d86/pybonjour-python3-e7c010db6372559b9ee9cb64dde75206db684d86.tar.gz"
+    sha256 "fd135e059b0a3ea42d9a4279e8b3fca04508cf13ecf627d9c11b58f2de1233f0"
+  end
+
+  resource "plugin_installer" do
+    url "https://ftp.gajim.org/plugins_1.1_zip/plugin_installer.zip"
+    sha256 "62f3c84060b4cc0f03bbc6b509b6db83831fa6f4ff977640f712a055424c2064"
+  end
+
+  resource "omemo" do
+    url "https://ftp.gajim.org/plugins_1.1_zip/omemo.zip"
+    sha256 "f1ed4690080cd10983135c206b8b5ef62b4aa28dd21a5cf1c294f59d272e2ed6"
+  end
+
+  resource "url_image_preview" do
+    url "https://ftp.gajim.org/plugins_1.1_zip/url_image_preview.zip"
+    sha256 "45a913f799ac6eff435fb738d95d6d2690c4a981704ee7b43c248b3d7b56984e"
+  end
+
+  def install
+    xy = Language::Python.major_minor_version "python3"
+    ENV["SDKROOT"] = MacOS.sdk_path if MacOS.version == :sierra
+
+    venv = virtualenv_create(libexec, "python3")
+
+    (resources.map(&:name).to_set - %w[entrypoints Pillow plugin_installer omemo url_image_preview]).each do |r|
+      venv.pip_install resource(r)
+    end
+    resource("entrypoints").stage do
+      # Without removing this file, `pip` will ignore the `setup.py` file and
+      # attempt to download the [`flit`](https://github.com/takluyver/flit)
+      # build system.
+      rm_f "pyproject.toml"
+      venv.pip_install Pathname.pwd
+    end
+    resource("Pillow").stage do
+      inreplace "setup.py" do |s|
+        sdkprefix = MacOS.sdk_path_if_needed ? MacOS.sdk_path : ""
+        s.gsub! "ZLIB_ROOT = None", "ZLIB_ROOT = ('#{sdkprefix}/usr/lib', '#{sdkprefix}/usr/include')"
+      end
+
+      venv.pip_install Pathname.pwd
+    end
+
+    venv.pip_install_and_link buildpath
+
+    %w[plugin_installer omemo url_image_preview].each do |r|
+      resource(r).stage do
+        (libexec/"lib/python#{xy}/site-packages/gajim/data/plugins").install Pathname.pwd
+      end
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/gajim --version")
```

Comment:
Can we get a better test here please?

Reply:
I'm afraid I don't know what else to test, the other options require graphical input and/or an already running instance.